main/src/library/RedBlackTree.flix

```diff
@@ -378,23 +378,18 @@ namespace RedBlackTree {
     /// Traverses the tree `t` in parallel.
     ///
     @Parallel
-    pub def parExists(n: Int32, f: (k, v) -> Bool, t: RedBlackTree[k, v]): Bool \ IO =
+    pub def parExists(n: Int32, f: (k, v) -> Bool, t: RedBlackTree[k, v]): Bool =
         if (n <= 1)
             seqExists(f, t)
         else
             match t {
                 case Node(_, a, k, v, b) =>
-                    let (sL, rL) = Channel.buffered(1);
-                    let (sR, rR) = Channel.buffered(1);
                     if (f(k, v))
                         true
-                    else {
-                        spawn (Channel.send(parExists((n - 2) / 2, f, a), sL)); // We divide the rest of the threads as follows:
-                        spawn (Channel.send(parExists((n - 2) / 2, f, b), sR)); // We spawn two new threads leaving us with n - 2
-                        let a1 = Channel.recv(rL);                              // that we distribute over the two spanned threads.
-                        let b1 = Channel.recv(rR);
-                        a1 or b1
-                    }
+                    else
+                        par (left <- parExists((n - 2) / 2, f, a); // We divide the rest of the threads as follows:
+                            right <- parExists((n - 2) / 2, f, b)) // We spawn two new threads leaving us with n - 2
```

Comment:
@mlutze and I know what you did....

Reply:
You could not help yourself, you really wanted those arrows to align :D

Reply:
haha yeah but it's also the default indentation! :wink:

Reply:
how about `l` and `r`?